homa_incoming.c

```diff
@@ -681,14 +681,31 @@ void homa_resend_pkt(struct sk_buff *skb, struct homa_rpc *rpc,
 			rpc->id, ntohl(h->offset), ntohl(h->length),
 			h->priority);
 
-	if (!homa_is_client(rpc->id)) {
+	if (!homa_is_client(rpc->id) && rpc->state != RPC_OUTGOING) {
 		/* We are the server for this RPC. */
-		if (rpc->state != RPC_OUTGOING) {
+		if ((rpc->msgin.length - rpc->msgin.bytes_remaining) >= rpc->msgin.granted) {
+			/* We've received everything that we've granted, so either the server 
+			   is busy with an influx of messages exceeding max_incoming, or a 
+			   grant packet has been dropped. In either case, the server should 
+			   initiate RESEND
```

Comment:
This comment doesn't look right: a BUSY gets sent here, not RESEND.

Reply:
By "initiate" resend I meant the server should be the one who issues the RESEND. I have changed the wording as I see how it's ambiguous.